src/shogun/statistical_testing/internals/Block.cpp

```diff
@@ -47,32 +46,33 @@ Block::Block(CFeatures* feats, index_t index, index_t size) : m_feats(feats)
 	SG_REF(m_feats);
 
 	// create a shallow copy and subset current block separately
-	CFeatures* block=FeaturesUtil::create_shallow_copy(feats);
-	ASSERT(block->ref_count()==0);
+	CFeatures* block=feats->shallow_subset_copy();
```

Comment:
shallow_subset_copy?
Why not shallow_copy and if there is a subset, it is copied?